File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let arr = [];
    for(let i=1; i<=n; i++) {
        if(n % i === 0 && i % 2 === 1) {
            arr.push(i)
        }
    }
    return arr.length
    // let answer = 0;
    // for (let i = 1; i <= n; i++) {
    //     if (n % i === 0 && i % 2 === 1) answer++;
    // }
    // return answer;

    
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0  # 연속된 자연수로 n을 표현하는 방법의 수
    for i in range(1, n+1):
        hap = 0
        
        # hap이 n보다 크거나 같아질 때까지 i부터 순차적으로 더함
        while hap < n:
            hap += i
            i += 1
        if hap == n: # 만약 hap이 n과 같으면 연속된 자연수로 나타낼 수 있으므로 방법 +1
            answer += 1
            
    return answer